src/main/java/org/checkerframework/checker/builder/qual/CalledMethodsPredicate.java

```diff
@@ -0,0 +1,34 @@
+package org.checkerframework.checker.builder.qual;
+
+import org.checkerframework.framework.qual.SubtypeOf;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * This annotation represents a predicate on @CalledMethods
+ * annotations that must be true. It is intended to extend
+ * the possible LUB of the @CalledMethods type system to
+ * permit methods to be annotated to require a disjunction
+ * of method calls. So, for instance, if it was acceptable
+ * to call method a() or method b() before calling method c(),
+ * then c()'s receiver should have an @CalledMethodsPredicate
+ * annotation with the argument "a || b"
+ *
+ * The argument is a string. The string must be constructed
+ * from the following grammar:
+ *
+ * S → method name | (S) | S && S | S || S
+ *
+ * That is, the permitted elements are method names, parentheses,
+ * and the strings "&&" and "||". "&&" has higher precedence
+ * than "||", following standard Java operator semantics.
```

Comment:
Have you thought about handling of method overloading?  I wouldn't be surprised if it comes up at some point.  Doesn't need to block this PR but probably we should have an open issue on it (as it will force us to go beyond just method names in these annotations)

Reply:
I had not thought about method overloading. I agree that there's no need to block this PR, but I'll open an issue and think about how to address it. Nice catch.